### ISSUE TITLE
FR: Auth key generation for Member user role

### ISSUE THREAD

**Issue by asad26** [tailscale/tailscale#6782]
### What are you trying to do?

According to the user roles document [User roles · Tailscale](https://tailscale.com/kb/1138/user-roles/#permissions-managed-by-user-roles), the user with “Member” role doesn’t have permission to generate Auth keys. We want to automate the process of joining the tailnet using Auth key while keeping the highest possible security (i.e., “Member” can authenticate by bypassing interactive login, without assigning any higher role that comes with a lot of permissions).

### How should we solve this?

It can be solved by providing the Auth key generation feature for the users with "Member" role. Is it something which can be changed / added so that Member’s also can generate the key and use it for login? 

### What is the impact of not solving this?

Currently, we are using the Auth key from a user with "Admin" role, which also comes with all the permissions. We don't want a normal user to perform any actions on the Tailscale account except logging in using Auth key.

**Comment by clarkm5** [comment]
Agreed - "member auth keys" could be defined in the ACL config which would also allow the specification of
- Whether the member can set the reusable, expiration, ephemeral, pre-authorized options
- Tags that should be applied to any machine authorised by the member auth key
- Group/user access to member auth key configs

Members can
- Set reusable, expiration, ephemeral, pre-auth depending on the ACL config
- Set additional Tags for which they are configured as tagOwners
- Generate an auth-key with the appropriate permissions/tags by selecting one of their permitted member auth key configurations

**Comment by TeaEwan** [comment]
On the `https://login.tailscale.com/admin/settings/keys` page, click "Generate auth key...", add the "select member user" function. This "auth key" will log in as that member user, not as an administrator. 

If possible, it will be very good. Thank you!

**Comment by IvanRibakov** [comment]
> Does that address the underlying problem you're solving here?

@mayakacz Not quite. Devices added using authkey generated via OAuth seem to have no expiry date compared to joining/reauthing via a link. IMO this difference defeats the purpose of automating the device (re-)authentication if they will not expire on their own.

**Comment by linuskendall** [comment]
This is a problem for us as well. We generally want users to have auth keys because we used them in codespaces creation so that the user can set up codespaces that can auto-join Tailscale with the correct permissions. For this to work, they'd need to specify auth key as a github secret.  However, this doesn't work since neither admins nor the users themselves can generate tokens for the user unless they're IT admin role or above.

**Comment by kpanchal360** [comment]
Agreed, have a requirement for member auth key generation ability.  As admin we are now on the hook for their key expiration creating unnecessary work and problems.
